README.md
# finalproject 
 
 1.delete the file /finalproject/routes.example.py
 2.go to web2py/example/routes.parametric.py and move that to your web2py folder
 3.rename routes.parametric.py to routes.py
 4. modify the routes.py as following:


 routers = dict(

    # base router
    
    BASE=dict(
        default_application='init',       #change this from welcome to init and add ALL for applications
              applications='ALL'
              

    ),
)


PS: I will modify this more to replace http://127.0.0.1:8000 with just our website name later thats easy, and also add either the cutome username or id number or however you guys want too later but for now once you goto the web page instead of showing 
http://127.0.0.1:8000/finalproject/default/index  if you are not logged in it will just show:
http://127.0.0.1:8000/finalproject and prompt you with a static home page 
just to welcome you which i just thought it looks nice and once you are logged in you direct to your profile with a route of:
http://127.0.0.1:8000/finalproject/profile_info + showing your first name and last name if you try to go to /profile_info while you are not logged in
you will redirect to log in page 

 


static/changelog.txt
== CHANGE LOG == 

= Version 1.3.4  = 27/10/2016
* Feature - Compatible with Woocommerce

= Version 1.3.3  = 14/10/2016
* Add - Added Options Backup function
* Fix - Fixed banner text responsive issue

= Version 1.3.2  = 30/09/2016
* Fix - Fixed 'Header Overlay' option issue

= Version 1.3.1  = 21/09/2016
* Add - Added 'Full Width Header' option
* Add - Added 'Overlay Header Text Color' option
* Add - Added 'Page Tilte-bar Text Color' option
* Add - Added 'Alchem Custom Menu' widget
* Add - Added slider pagination
* Fix - Fixed customize layout issue in Firefox

= Version 1.3.0  = 29/07/2016
* Add - Added "Color" option for "Banner" Section
* Add - Added "Icon Color" option for "Service" Section
* Add - Added lightbox for "Gallery" Section
* Add - Added "Category" option for "Latest news" section
* Fix - Fixed header overlay issue

= Version 1.2.9  = 21/07/2016
* Add - Added section 'Background Parallax Image' option

= Version 1.2.8  = 08/07/2016
* Fix - Fixed Blog Archive sidebar issue
* Fix - Fixed search page sidebar issue
* Fix - Fixed 404 page sidebar issue

= Version 1.2.7  = 07/07/2016
* Fix - Fixed missing Custom CSS issue
* Fix - Fixed page sidebar options issue

= Version 1.2.6  = 05/07/2016
* Add - Added banner video background function
* Add - Added 404 page sidebar options
* Add - Added home page > partner section tooltip
* Fix - Fixed Site Width option issue

= Version 1.2.5  = 24/06/2016
* Removed Sidebar > Pages options
* Removed Blog > Blog Single Post Page Options > Blog Post Sidebar Position option
* Removed Blog > General Blog Options > Blog Archive/Author/Category Sidebar option
* Fix - Fixed sidebar options issue
* Fix - Fixed footer social icons box color option issue

= Version 1.2.4  = 20/06/2016
* Add - Added Logo Position option
* Add - Added Blog List Style option
* Add - Added Page Background Meta Box option ( Display Title )
* Add - Added Page Background Meta Box option ( Background Image )
* Add - Added Page Background Meta Box option ( Background Repeat )
* Add - Added Title Bar Background Meta Box option ( Title Bar Background Image )
* Add - Added Title Bar Background Meta Box option ( Title Bar Background Repeat )
* Add - Added Post Meta Box option( Display Meta Data )
* Add - Added Post Meta Box option( Display Share Icons )
* Add - Added Post Meta Box option( Display Author Info )
* Add - Added Post Meta Box option( Display Related Posts )
* Fix - Fixed archive page title issue
* Fix - Fixed slider Autoplay option issue
* Fix - Fixed slider Slideshow Speed option issue

= Version 1.2.3  = 24/05/2016
* Fix - Fixed footer social icons box color issue
* Fix - Fixed parallax backgrounds issue
* Fix - Fixed footer widget area posts list background issue
* Fix - Fixed Tooltip Position for Social Icons issue

= Version 1.2.2  = 18/05/2016
* Section slogan button

= Version 1.2.1  = 08/04/2016
* Fix - Fixed Display Title Bar page option issue
* Fix - Fixed page Slider Banner issue

= Version 1.2.0  = 31/03/2016
* Fix - Fixed single slide issue of 'section banner'

= Version 1.1.9  = 28/03/2016
* Add - Added 2 home page styles
* Add - Added home page animation on/off option
* Fix - Fixed waypoints issue
* Fix - Fixed blog timeline issue
* Fix - Fixed waypoints issue
* Removed home page animation on mobile

= Version 1.1.8  = 16/03/2016
* Fixed full height of homepage banner in Firefox.

= Version 1.1.7  = 03/03/2016
* Added: home page slider ( Section - Banner)
* Added: home page section animation

= Version 1.1.6 = 01/03/2016
* Fix - Fixed home page section id option issue ( Section - Features )

= Version 1.1.5 = 26/02/2016
* Added 'Google Fonts' option
* Removed duplicate theme option

= Version 1.1.4 = 14/02/2016
* Fixed Social Icon Link options issue ( Section - Our Team )

= Version 1.1.3 =
* Fixed 'Copyright Background Color' options issue
* Fixed 'Nav Menu' metabox option issue
* Added 'Styling' options
* Added 'Main Menu Dropdown Width' option

= Version 1.1.1 =
* Added header style 2

= Version 1.1.0 =
* Fixed theme option issue: Blog\Blog Single post page option\Disable featured iamage
* Fixed copyright content HTML issue

= Version 1.0.9 =
* Added blog options

= Version 1.0.8 =
* Fixed home page sections content HTML issue

= Version 1.0.2 =
* Transfer data from theme options to customizer
* Fixed blog template sidebar issue
* Fixed blog pagingnav issue
* Fixed blog Excerpt Length issue

= Version 1.0.1 =
* Added bbpress style
* Fixed header padding issue
* Fixed page right sidebar issue
* Fixed top bar info color issue
* Fixed sticky header background color issue

models/menu.py
# -*- coding: utf-8 -*-
# this file is released under public domain and you can use without limitations

# ----------------------------------------------------------------------------------------------------------------------
# Customize your APP title, subtitle and menus here
# ----------------------------------------------------------------------------------------------------------------------

response.logo = A(B('web', SPAN(2), 'py'), XML('&trade;&nbsp;'),
                  _class="navbar-brand", _href="http://www.web2py.com/",
                  _id="web2py-logo")
response.title = request.application.replace('_', ' ').title()
response.subtitle = ''

# ----------------------------------------------------------------------------------------------------------------------
# read more at http://dev.w3.org/html5/markup/meta.name.html
# ----------------------------------------------------------------------------------------------------------------------
response.meta.author = myconf.get('app.author')
response.meta.description = myconf.get('app.description')
response.meta.keywords = myconf.get('app.keywords')
response.meta.generator = myconf.get('app.generator')

# ----------------------------------------------------------------------------------------------------------------------
# your http://google.com/analytics id
# ----------------------------------------------------------------------------------------------------------------------
response.google_analytics_id = None

# ----------------------------------------------------------------------------------------------------------------------
# this is the main application menu add/remove items as required
# ----------------------------------------------------------------------------------------------------------------------

response.menu = [
    (T('Home'), False, URL('default', 'index'), [])
]

DEVELOPMENT_MENU = True


# ----------------------------------------------------------------------------------------------------------------------
# provide shortcuts for development. remove in production
# ----------------------------------------------------------------------------------------------------------------------

def _():
    # ------------------------------------------------------------------------------------------------------------------
    # shortcuts
    # ------------------------------------------------------------------------------------------------------------------
    app = request.application
    ctr = request.controller

    # ------------------------------------------------------------------------------------------------------------------
    # useful links to internal and external resources
    # ------------------------------------------------------------------------------------------------------------------
    response.menu += [
        (T('My Sites'), False, URL('admin', 'default', 'site')),
        (T('This App'), False, '#', [
            (T('Design'), False, URL('admin', 'default', 'design/%s' % app)),
            LI(_class="divider"),
            (T('Controller'), False,
             URL(
                 'admin', 'default', 'edit/%s/controllers/%s.py' % (app, ctr))),
            (T('View'), False,
             URL(
                 'admin', 'default', 'edit/%s/views/%s' % (app, response.view))),
            (T('DB Model'), False,
             URL(
                 'admin', 'default', 'edit/%s/models/db.py' % app)),
            (T('Menu Model'), False,
             URL(
                 'admin', 'default', 'edit/%s/models/menu.py' % app)),
            (T('Config.ini'), False,
             URL(
                 'admin', 'default', 'edit/%s/private/appconfig.ini' % app)),
            (T('Layout'), False,
             URL(
                 'admin', 'default', 'edit/%s/views/layout.html' % app)),
            (T('Stylesheet'), False,
             URL(
                 'admin', 'default', 'edit/%s/static/css/web2py-bootstrap3.css' % app)),
            (T('Database'), False, URL(app, 'appadmin', 'index')),
            (T('Errors'), False, URL(
                'admin', 'default', 'errors/' + app)),
            (T('About'), False, URL(
                'admin', 'default', 'about/' + app)),
        ]),
        ('web2py.com', False, '#', [
            (T('Download'), False,
             'http://www.web2py.com/examples/default/download'),
            (T('Support'), False,
             'http://www.web2py.com/examples/default/support'),
            (T('Demo'), False, 'http://web2py.com/demo_admin'),
            (T('Quick Examples'), False,
             'http://web2py.com/examples/default/examples'),
            (T('FAQ'), False, 'http://web2py.com/AlterEgo'),
            (T('Videos'), False,
             'http://www.web2py.com/examples/default/videos/'),
            (T('Free Applications'),
             False, 'http://web2py.com/appliances'),
            (T('Plugins'), False, 'http://web2py.com/plugins'),
            (T('Recipes'), False, 'http://web2pyslices.com/'),
        ]),
        (T('Documentation'), False, '#', [
            (T('Online book'), False, 'http://www.web2py.com/book'),
            LI(_class="divider"),
            (T('Preface'), False,
             'http://www.web2py.com/book/default/chapter/00'),
            (T('Introduction'), False,
             'http://www.web2py.com/book/default/chapter/01'),
            (T('Python'), False,
             'http://www.web2py.com/book/default/chapter/02'),
            (T('Overview'), False,
             'http://www.web2py.com/book/default/chapter/03'),
            (T('The Core'), False,
             'http://www.web2py.com/book/default/chapter/04'),
            (T('The Views'), False,
             'http://www.web2py.com/book/default/chapter/05'),
            (T('Database'), False,
             'http://www.web2py.com/book/default/chapter/06'),
            (T('Forms and Validators'), False,
             'http://www.web2py.com/book/default/chapter/07'),
            (T('Email and SMS'), False,
             'http://www.web2py.com/book/default/chapter/08'),
            (T('Access Control'), False,
             'http://www.web2py.com/book/default/chapter/09'),
            (T('Services'), False,
             'http://www.web2py.com/book/default/chapter/10'),
            (T('Ajax Recipes'), False,
             'http://www.web2py.com/book/default/chapter/11'),
            (T('Components and Plugins'), False,
             'http://www.web2py.com/book/default/chapter/12'),
            (T('Deployment Recipes'), False,
             'http://www.web2py.com/book/default/chapter/13'),
            (T('Other Recipes'), False,
             'http://www.web2py.com/book/default/chapter/14'),
            (T('Helping web2py'), False,
             'http://www.web2py.com/book/default/chapter/15'),
            (T("Buy web2py's book"), False,
             'http://stores.lulu.com/web2py'),
        ]),
        (T('Community'), False, None, [
            (T('Groups'), False,
             'http://www.web2py.com/examples/default/usergroups'),
            (T('Twitter'), False, 'http://twitter.com/web2py'),
            (T('Live Chat'), False,
             'http://webchat.freenode.net/?channels=web2py'),
        ]),
    ]


if DEVELOPMENT_MENU:
    _()

if "auth" in locals():
    auth.wikimenu()
